src/runtime_src/core/edge/user/device_linux.cpp

```diff
@@ -157,9 +157,18 @@ struct dev_info
   }
 };
 
+struct aie_metadata_info{
+  uint32_t num_cols;
+  uint32_t num_rows;
+  uint32_t shim_row;
+  uint32_t core_row;
+  uint32_t mem_row;
+  uint32_t num_mem_row;
+};
+
 // Function to get aie max rows and cols by parsing aie_metadata sysfs node
-static void
-get_aie_row_col(const xrt_core::device* device, uint32_t &row, uint32_t &col)
+static void 
+get_aie_row_col(const xrt_core::device* device, aie_metadata_info &aie_meta)
```

Comment:
```suggestion
get_aie_row_col(const xrt_core::device* device, aie_metadata_info& aie_meta)
```
The function name indicates a return value.  If aie_metadata_info is appended, the the function name should be `add_aie_row_col`.   If aie_metadata_info is completely filled by this function, then it should be constructed and returned by this function.